### PR TITLE
M2 r cleanup

### LINKED ISSUE #564: [Refactor]: Remove the magrittr pipe and use native pipe instead
### Refactor request

R 4.1.0 provides the native pipe and there is no longer a need for magrittr::%>%. [The tidyverse blog explains the differences](https://www.tidyverse.org/blog/2023/04/base-vs-magrittr-pipe/) but given R is now on version 4.4.0 I think we do not need backward compatibility and can remove the code below after changing all magrittr pipes to native pipes.

https://github.com/NOAA-FIMS/FIMS/blob/97281c20ba697c66cbbbf5fe4f01241f79183676/R/utils-pipe.R#L1-L14

### LINKED ISSUE #567: [Refactor]: R code does not adhere to style guide
### Refactor request

The R code in the FIMS repository has code that deviates from the style guide recommended in the collaborative workflow even though the code is being styled by a styler. 

### Tasks
- [x] parameter and object names should be snake_case, e.g., `n_years` not `nyears` in fimsframe.R and test-fimsframe.R (@MOshima-PIFSC)
- [x] determine if the usethis package should be suggested or imported (test moving it to suggests in a branch)
- ~~look for additional rules to add to the styler to catch these types of deviations~~
- [x] open an issue in the case-studies repo about removing FIMSFrameAge from case studies
- [x] #564 
- [x] #590 
- [x] function names should align with the guide rather than be a mix of snake_case and camelCase; issue NOAA-FIMS/collaborative_workflow#133 will need to be decided upon before this task can more forward
- [x] roxygen @params should be sentences with subsequent lines indented with two spaces.
- [x] roxygen titles should be less than 80 but preferably 60 characters in sentence case without a full stop
- [x] roxygen even though I advocated for it, it has not officially accepted to use {} around package names
- [x] roxygen paragraph entries should start on a new line, e.g., @description, @source, @format, and @return should not have text on the same line
- [x] roxygen R/data_mile1.R#L6 can be wrapped to be less than 80 characters per line
- [x] #565
- [x] `use_template` and `ui_stop` are being imported in [R/FIMS-package.R](https://github.com/NOAA-FIMS/FIMS/blob/97281c20ba697c66cbbbf5fe4f01241f79183676/R/FIMS-package.R#L8) but they are being called with `::` so there is no need to import them
- [x] use `utils::head()` rather than importing it
- [x] Remove other unused functions should be removed from the code base

### Path forward

@MOshima-PIFSC and @kellijohnson-NOAA will work on this in the m2-r-cleanup branch, where tasks will be completed in sub-feature branches, e.g., m2-r-cleanup-zzz, that are merged into the feature branch daily. Then a pull request will be submitted after style-r is rebased to main. We can work on one file at a time rather than one task at a time, then move onto the tasks that cross multiple files. The timeline for completion is May 17th, 2024.

### LINKED ISSUE #588: [Developer Issue]: are use_module() and create_rcpp_interface_object() being used?
### Issue details

I am wondering if `use_module()` and `create_rcpp_interface_object()` are being used by anyone? They are both developer functions rather than user functions so it makes sense that they are not in the code base but given the limited scope/functionality of `use_module()` I think it should be removed. @msupernaw is `create_rcpp_interface_object()` up to date because I know it was created a while ago?

At a minimum, I think that these functions, if kept, should not be exported.

### LINKED ISSUE #590: [Refactor]: Remove FIMSFrameAge
### Refactor request

Remove FIMSFrameAge because the complexity is not needed and it causes confusion. This will require moving some functionality from FIMSFrameAge to FIMSFrame and removing FIMSFrameAge from the vignette and all case studies that copied that part of the vignette into their quarto documents.

### PR DESCRIPTION
**Pull request by kellijohnson-NOAA**
# What is the feature?

@MOshima-PIFSC and @kellijohnson-NOAA worked on cleaning up the R code, which turned into decisions about what functions actually needed to be kept. Then, subsequent clean up of what was left.
* General clean up of all files in /R/ folder and associated tests and vignettes.
* Attempt to adhere to tidyverse style guide for R code, e.g., snake_case.
* More consistent formatting of roxygen tags.
* Removal of unused functions.
* After removing unused function, it appeared that usethis was no longer needed. We tested this in a branch and all tests passed so it the movement of usethis to Suggests is also included in this PR.

Close #567 
Close #590
Close #588
Close #564
See a more detailed list of changes that were attempted to include in this PR in #567.

# How have you implemented the solution?

# Does the PR impact any other area of the project?

## Input
There is an [associated issue in the case studies](https://github.com/NOAA-FIMS/case-studies/issues/25) about removing the use of FIMSFrameAge and just using FIMSFrame that @iantaylor-NOAA is in charge of and will be implementing once this PR is approved. @iantaylor-NOAA note, we also changed to `n_years`, `n_ages`, `weight_at_age`, etc. for object names such that there are no abbreviations and everything is snake case.

## Output
NA

# How to test this change

# Developer pre-PR checklist

- [x] I relied on GitHub actions to :test_tube: things for me while I sat on the :couch_and_lamp:.

@jbigman-NOAA I assigned you as a reviewer because these changes are rather benign and more about general formatting and I thought it might help you jump into the code. Also, I am excited for a fresh perspective/thoughts about the R code in FIMS. Feel free to let me know if you are not interested or do not have time to review this PR.

### CODE REVIEW

File: R/fimsframe.R
Comment (Bai-Li-NOAA) [inline R/fimsframe.R:67]: Question: I remember m_*() means module, do we want to explicitly write it out in the code (e.g., module_weight_at_age) or add a line of documentation in this file where it was initially introduced?

File: pkgdown/_pkgdown.yml
Comment (Bai-Li-NOAA) [inline pkgdown/_pkgdown.yml:40]: Suggestion: remove two functions (i.e., create_fims_rcpp_interface and use_module) under the "Package development" section

**Review by Bai-Li-NOAA — CHANGES_REQUESTED** [review]
Great work! I have one minor request to update the `_pkgdown.yml`. I have also created a new issue (#595) to render the pkgdown website on pull requests to make sure that the pkgdown website can be built before merging changes into the main branch in the future. Let me know if you have any questions.

File: pkgdown/_pkgdown.yml
Comment (kellijohnson-NOAA) [reply pkgdown/_pkgdown.yml]: Thanks for catching this @Bai-Li-NOAA, I have fixed the commit that this was a part of and it is now in the PR.

File: R/fimsframe.R
Comment (kellijohnson-NOAA) [reply R/fimsframe.R:67]: I think we should leave it as is until we decide how to "hide" these `m_*()` functions and just make them happen internally rather than how they are being used now within the case studies.

**Comment by kellijohnson-NOAA** [comment]
@Bai-Li-NOAA I am thinking that we should squash all of these commits into a single commit before we merge it into main. What are your thoughts? I can do this and force push or we can just use the squash and merge button if you agree.

**Comment by Bai-Li-NOAA** [comment]
@kellijohnson-NOAA Good idea! It would be great if you can do it and force push. Thank you!

**Comment by kellijohnson-NOAA** [comment]
Okay @Bai-Li-NOAA it is now a single commit that can be merged in once the tests are passing. Who does the merge, the reviewer or the committer?

**Comment by Bai-Li-NOAA** [comment]
@kellijohnson-NOAA. The committer merges the changes.

**Pull request merged**
